src/object/srv_obj.c

```diff
@@ -648,9 +773,42 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 	if (opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_UPDATE ||
 	    opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_TGT_UPDATE) {
 		bulk_op = CRT_BULK_GET;
+		if (oca->ca_resil == DAOS_RES_EC) {
+			unsigned int	tgt_idx = orw->orw_oid.id_shard -
+							orw->orw_start_shard;
+
+			for (i = 0; i < orw->orw_nr; i++)
+				skip_list[i] = NULL;
+
+			if (tgt_idx >= oca->u.ec.e_p) {
+				rc = ec_data_target(tgt_idx - oca->u.ec.e_p,
+						    orw->orw_nr, tmp_iods,
+						    oca, skip_list);
```

Comment:
these seems quite a lot processing and also complex on server side. so I wonder whether we should do it on client side (like ec encoding). We can discuss later.
Btw: this part along with those split/encoding part probably needs a separate unit test, instead of only testing it by the whole stack.

Reply:
I ran unit tests on the server code, and I also have a unit test for the client side.  I also did testing on the whole stack.  I will open a branch and build a subsequent pull request to get this testing into the build (need to learn about cmocka first).